inkcpp/include/globals.h

```diff
@@ -4,15 +4,43 @@
 
 namespace ink::runtime
 {
+	template<typename T>
+	struct ret {
+		operator T() { return value; }
+		operator bool() { return has_value; }
+		T value;
+		bool has_value;
+	};
 	/**
 	* Represents a global store to be shared amongst ink runners.
 	* Stores global variable values, visit counts, turn counts, etc.
 	*/
-	class globals_interface
+	class globals
```

Comment:
You should keep this as `globals_interface`. `globals` is defined in `include\types.h` as a shared pointer to `globals_interface`.

Reply:
ok :)